libsubid/api.c

```diff
@@ -38,6 +38,25 @@
 #include "idmapping.h"
 #include "subid.h"
 
+char *Prog = "(libsubid)";
+extern int shadow_logfd;
+
+void libsubid_init(const char *progname, FILE * logfd)
+{
+	if (progname)
+		Prog = progname;
+
+	if (logfd) {
+		shadow_logfd = logfd;
+		return;
+	}
+	shadow_logfd = fopen("/dev/null", O_RDWR);
+	if (!shadow_logfd) {
+		fprintf(stderr, "ERROR opening /dev/null for error messages.  Using stderr.");
```

Comment:
I'm not sure this is expected if caller explicitly requested "dev/null".
Perhaps error return code for init() function would make more sense here.

Reply:
I'll return false on error to help the library figure out what's going on.

Reply:
Thanks for the update with error code.

I think `return false` should be enough here.
IMO, `fprintf()` in this case is unnecessary and maybe even undesirable (taking into account "/dev/null" was requested explicitly).